pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +27,115 @@ import (
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:scope=Namespaced
 type StorageCohort struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a cohort.
+	// +required
+	Spec StorageCohortSpec `json:"spec"`
+
+	// Most recently observed status of the cohort.
+	// Populated by the cohort operator or cohort manager.
+	// +optional
+	Status StorageCohortStatus `json:"status,omitempty"`
+}
+
+// StorageCohortSpec describes the attributes that a cohort is created with.
+type StorageCohortSpec struct {
+	// NodeSelector is used to specify the cohort to be considered
+	// +required
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector"`
+
+	// CohortManager stores all the details about the cohort manager responsible for
+	// managing the cohort
+	// +optional
+	CohortManager []map[string]string `json:"cohortManager,omitempty"`
```

Comment:
Shall we make this simply map[string]string ? any reason for requiring the list?

Reply:
There was a user requirement (from the design doc) for specifying multiple URLs for the CohortManager. However, we can achieve the same within the same map as well, using delimiters or different keys. So, we will be taking up this review comment.